genesis/ext/pyrender/utils.py

```diff
@@ -87,15 +87,21 @@ def format_texture_source(texture, target_channels="RGB"):
             if texture.shape[2] == 1:
                 texture = np.repeat(texture, 3, axis=2)
             elif texture.shape[2] == 2:
-                raise ValueError("Cannot reformat 2-channel texture into RGB")
+                # Convert 2-channel texture to grayscale RGB by duplicating the first channel
+                texture = texture[:, :, (0, 0, 0)]
             else:
                 texture = texture[:, :, (0, 1, 2)]
         elif target_channels == "RGBA":
             if texture.shape[2] == 1:
                 texture = np.repeat(texture, 4, axis=2)
                 texture[:, :, 3] = 255
             elif texture.shape[2] == 2:
-                raise ValueError("Cannot reformat 2-channel texture into RGBA")
+                luminance = texture[:, :, 0]
+                alpha = texture[:, :, 1]
+                rgba = np.empty((texture.shape[0], texture.shape[1], 4), dtype=texture.dtype)
+                rgba[:, :, :3] = luminance[:, :, None]
+                rgba[:, :, 3] = alpha
+                texture = rgba
```

Comment:
What about just calling it `texture` at this point ? You already exacted the values so it should be fine.

Reply:
No, because of numpy broadcasting.